Plutarch/DataRepr.hs

```diff
@@ -22,5 +24,27 @@ module Plutarch.DataRepr (
   F.HRec,
 ) where
 
+import Plutarch
+import Plutarch.Builtin
 import qualified Plutarch.DataRepr.Internal as I
 import qualified Plutarch.DataRepr.Internal.Field as F
+import Plutarch.Internal
+import Plutarch.List
+
+{- | Cons a field to a data record.
+
+You can specify the label to associate with the field using type applications-
+
+@
+
+foo :: Term s (PDataRecord '[ "fooField" ':= PByteString ])
+foo = pdcons @"fooField" # pdata (phexByteStr "ab") # pdnil
+
+@
+-}
+pdcons :: forall label a l s. Term s (PAsData a :--> I.PDataRecord l :--> I.PDataRecord ((label 'I.:= a) ': l))
```

Comment:
I wasn't sure about the type variable ordering here. Should `label` come before or after `a` - which one would users be more likely to use type application for?